server/testingHelpers.js

```diff
@@ -94,14 +94,14 @@ function sessionFactory(overrides = {}) {
   // prettier-ignore
   return new Session(
     overrides.id !== undefined ? overrides.id : 'd91593b4-25ce-11ec-9621-0242ac130002',
-    overrides.locationid !== undefined ? overrides.locationid : 'myLocation',
     overrides.chatbotState !== undefined ? overrides.chatbotState : CHATBOT_STATE.COMPLETED,
     overrides.alertType !== undefined ? overrides.alertType : ALERT_TYPE.SENSOR_STILLNESS,
     overrides.createdAt !== undefined ? overrides.createdAt : new Date('2021-10-05T20:20:20.000Z'),
     overrides.updatedAt !== undefined ? overrides.updatedAt : new Date('2021-10-05T20:20:55.000Z'),
     overrides.incidentCategory !== undefined ? overrides.incidentCategory : 'Overdose',
     overrides.respondedAt !== undefined ? overrides.respondedAt : new Date('2021-10-05T20:20:33.000Z'),
     overrides.respondedByPhoneNumber !== undefined ? overrides.respondedByPhoneNumber : '+17778886666',
+    overrides.location !== undefined ? overrides.location : locationFactory(),
   )
 }
 
```

Comment:
I assume the change to the `Session` class to include more information on the location than just the id is necessary for the `resetStillnessTimer` to work? Could you explain which fields are needed?

Reply:
Yes, you're right! I did need some more information on the location. Specifically, I needed the `radarCoreId`. In `BraveAlerterConfigurator`, the code is now:
```
           const oldStillnessTimer = await particleFunctions.resetStillnessTimer(
              session.location.radarCoreId,
              helpers.getEnvVar('PARTICLE_PRODUCT_GROUP'),
            )
            if (oldStillnessTimer > -1) {
              helpers.log(`Reset stillness timer for ${session.location.locationid} from ${oldStillnessTimer} to 0`)
            } else {
              helpers.log(`Did not reset stillness timer for ${session.location.locationid}`)
            }
```

Instead of changing the `Session` class, I could have just added a new DB call in `BraveAlerterConfigurator` to `db.getLocationData(session.locationId)` and that would have worked too and required fewer code changes.

However, I've been slowly moving our DB model classes over to this pattern of having the full referenced object retrieved and not just IDs. You'll see how the `Location` class in `Location.js` as a whole `client`, not just its ID. And same for `SensorsVital`. Similarly, the `Session` class in Buttons has the whole `button` object, not just its id.

It's a tradeoff. This style causes more DB calls than is probably necessary, but makes it easier to read and write the JS code. A lot of frameworks and languages offer something along these lines. We just do it ourselves, but things like Django for Python do it (and migration scripts) under the hood. It falls under the idea of object-relational-mapper (ORM) if you'd like to learn more (https://www.theserverside.com/definition/object-relational-mapping-ORM).

Reply:
Cool! No problem with that. Thanks for explaining!